### DISCUSSION TITLE
Add TypeScript twoslash query-like functionality to Helix

### DISCUSSION THREAD

**Discussion by invakid404** [helix-editor/helix#6437] [Ideas]
There's this one VSCode extension that I really like using that's called [vscode-twoslash-queries](https://github.com/orta/vscode-twoslash-queries) that lets you use comments of the form `// ^?` to show you inline type information. In the case of the twoslash query extension, it's a TypeScript-specific thing. It's also a feature of the official TypeScript playground.

I'm thinking it would be useful if Helix supported this in the general case. My particular use case would be debugging complicated types and observing how they change in real-time. Being able to look at the inferred type for multiple type expressions at the same time and being able to see how my changes affect them without having to manually check them one by one saves me a lot of time and is what I use the twoslash queries extension in VSCode for.

This sounds like it could be generalized into a language-agnostic feature, although I wouldn't know if there's a valid use case outside of TypeScript, except I guess for "illustrative" purposes.

This is how it looks in VSCode (and the TypeScript playground):
![image](https://user-images.githubusercontent.com/38114607/227720790-dabdd3b2-806f-49ea-8773-be999443c497.png)

**Comment by the-mikedavis** [comment]
It sounds like this is already covered by two LSP features. You can use `<space>k` with your cursor over a type to get the type information from the language server. #5934 was also recently merged which shows type information as virtual text in the code.
